### PR TITLE
SKX AVX512 fixes

### PR DESCRIPTION
**Pull request by deeptiag1**
Still in testing.

### CODE REVIEW

File: src/generator_gemm_sse_avx_avx2_avx512.c [hunk 1]
```diff
@@ -644,9 +644,8 @@ LIBXSMM_API_INTERN unsigned int libxsmm_generator_gemm_sse_avx_avx2_avx512_get_i
         l_use_masking_a_c = 1;
       }
     }
-  } else if ( ( i_arch <= LIBXSMM_X86_AVX512_VL256 ) && ( ( LIBXSMM_GEMM_PRECISION_I8  == LIBXSMM_GETENUM_INP( i_xgemm_desc->datatype ) )  ||
+  } else if ( ( i_arch <= LIBXSMM_X86_AVX512_VL256) && ( ( LIBXSMM_GEMM_PRECISION_I8  == LIBXSMM_GETENUM_INP( i_xgemm_desc->datatype ) )  ||
```
Comment (alheinecke) [inline src/generator_gemm_sse_avx_avx2_avx512.c:647]: Wouldn't it be better to also incl. > AVX2

File: src/generator_gemm_sse_avx_avx2_avx512.c [hunk 2]
```diff
@@ -656,6 +655,18 @@ LIBXSMM_API_INTERN unsigned int libxsmm_generator_gemm_sse_avx_avx2_avx512_get_i
         l_use_masking_a_c = 1;
       }
     }
+  } else if ( ( i_arch <= LIBXSMM_X86_AVX512_CORE ) && ( ( LIBXSMM_GEMM_PRECISION_I8  == LIBXSMM_GETENUM_INP( i_xgemm_desc->datatype ) )  ||
```
Comment (alheinecke) [inline src/generator_gemm_sse_avx_avx2_avx512.c:658]: wouldn't it be better to also incl >=AVX512

File: src/generator_gemm_sse_avx_avx2_avx512.c [hunk 3]
```diff
@@ -844,7 +855,7 @@ LIBXSMM_API_INTERN unsigned int libxsmm_generator_gemm_sse_avx_avx2_avx512_updat
     } else {
       /* we are done with m_blocking */
     }
-  } else if ( ( i_arch <= LIBXSMM_X86_AVX512_VL256 ) && ( ( LIBXSMM_GEMM_PRECISION_I8  == LIBXSMM_GETENUM_INP( i_xgemm_desc->datatype ) )  ||
+  } else if ( ( i_arch <= LIBXSMM_X86_AVX512_VL256) && ( ( LIBXSMM_GEMM_PRECISION_I8  == LIBXSMM_GETENUM_INP( i_xgemm_desc->datatype ) )  ||
```
Comment (alheinecke) [inline src/generator_gemm_sse_avx_avx2_avx512.c:858]: see above

File: src/generator_gemm_sse_avx_avx2_avx512.c [hunk 4]
```diff
@@ -855,6 +866,18 @@ LIBXSMM_API_INTERN unsigned int libxsmm_generator_gemm_sse_avx_avx2_avx512_updat
     } else {
       /* we are done with m_blocking */
     }
+  } else if ( ( i_arch <= LIBXSMM_X86_AVX512_CORE ) && ( ( LIBXSMM_GEMM_PRECISION_I8  == LIBXSMM_GETENUM_INP( i_xgemm_desc->datatype ) )  ||
```
Comment (alheinecke) [inline src/generator_gemm_sse_avx_avx2_avx512.c:869]: see above

**Review by alheinecke — CHANGES_REQUESTED** [review]
I'm still not sure that fixes the issues as this not around the VNNI instructions...

**Comment by deeptiag1** [comment]
After this fix, 512 code paths work fine for integers.

File: src/generator_gemm_sse_avx_avx2_avx512.c [hunk 5]
```diff
@@ -855,6 +866,18 @@ LIBXSMM_API_INTERN unsigned int libxsmm_generator_gemm_sse_avx_avx2_avx512_updat
     } else {
       /* we are done with m_blocking */
     }
+  } else if ( ( i_arch > LIBXSMM_X86_AVX512_VL256) && ( i_arch <= LIBXSMM_X86_AVX512_CORE ) && ( ( LIBXSMM_GEMM_PRECISION_I8  == LIBXSMM_GETENUM_INP( i_xgemm_desc->datatype ) )  ||
```
Comment (alheinecke) [inline src/generator_gemm_sse_avx_avx2_avx512.c:869]: hmmm this covers also VL256 CLX... this sounds strange as it doesn't cover 512 CLX

File: src/generator_gemm_sse_avx_avx2_avx512.c [hunk 6]
```diff
@@ -656,6 +655,18 @@ LIBXSMM_API_INTERN unsigned int libxsmm_generator_gemm_sse_avx_avx2_avx512_get_i
         l_use_masking_a_c = 1;
       }
     }
+  } else if ( ( i_arch > LIBXSMM_X86_AVX512_VL256) && ( i_arch <= LIBXSMM_X86_AVX512_CORE ) && ( ( LIBXSMM_GEMM_PRECISION_I8  == LIBXSMM_GETENUM_INP( i_xgemm_desc->datatype ) )  ||
```
Comment (alheinecke) [inline src/generator_gemm_sse_avx_avx2_avx512.c:658]: hmmm this covers also VL256 CLX... this sounds strange as it doesn't cover 512 CLX

**Review by alheinecke — CHANGES_REQUESTED** [review]
Some conditions seems be still missing as this handles blocking, but still never (even under sub-optimal blocking) VNNI code should be generated on SKX.

File: src/generator_gemm_sse_avx_avx2_avx512.c (hunk 6 above)
Comment (deeptiag1) [reply src/generator_gemm_sse_avx_avx2_avx512.c]: Yes , this was a bad condition. Fixed it.

File: src/generator_gemm_sse_avx_avx2_avx512.c (hunk 5 above)
Comment (deeptiag1) [reply src/generator_gemm_sse_avx_avx2_avx512.c]: Yes , this was a bad condition. Fixed it.

**Review by alheinecke — APPROVED** [review]

**Pull request merged**